daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -905,13 +907,26 @@ object Main extends Logging {
                     }
                   } else {
                     // not streaming, show left over data warning
+                    val Dump = new DataDumper
+                    var (dataText, dataHex) = ("", "")
+                    var curPositionInByte0b = inStream.inputSource.position
+                    var bytesAvailable = inStream.inputSource.bytesAvailable
+                    var bytesLimit = math.min(8, bytesAvailable).toInt
```

Comment:
I think all these var's can be made vals. You'll have to slightly change how dataText and dataHex are set.

Reply:
Changed made